src/odemis/acq/move.py

```diff
@@ -875,40 +923,81 @@ def _doCryoSwitchSamplePosition(future, target):
                 raise CancelledError()
             future._task_state = FINISHED
 
-# Note: this transformation consists of translation of along x and y
-# axes, and 7 degrees rotation around rx, and 180 degree rotation around rz.
-# The rotation angles are constant existing in "FM_POS_ACTIVE" metadata,
-# but the translation are calculated based on the current position and some
-# correction/shifting parameters existing in metadata "FM_POS_ACTIVE". 
-# This correction parameters can change every session. They are calibrated
-# at the beginning of each run.
+
+def check_keys_in_md_calib(calibrated_values: dict) -> bool:
+    """
+    Check the required keys in the calibrated values.
+    :param calibrated_values:
+    :return: True if all keys are in the calibrated values, False otherwise.
+    """
+    required_keys = {'x', 'y', 'm', 'z', 'z_ct', 'dx', 'dy'}
+    if not required_keys.issubset(calibrated_values.keys()):
+        raise ValueError("Calibrated parameter does not have all required keys %s." % required_keys)
+    return True
+
+
+# Note: this transformation consists of translation and rotation.
+# The translations are along the x, y and m axes. They are calculated based on
+# the current position and some calibrated values existing in metadata "CAL_VAL".
+# The rotations are 180 degree around the rm axis, and a calibrated angle around the rx axis.
+# These angles exist in the metadata "FM_POS_ACTIVE".
 def transformFromSEMToMeteor(pos, stage):
     """
     Transforms the current stage position from the SEM imaging area to the 
         meteor/FM imaging area.
-    pos (dict str->float): the current stage position. The position has to have x, y, rz, and rx axes,
+    pos (dict str->float): the current stage position. The position has to have x, y, m, z, rx and rm axes,
         otherwise error would be raised.
     stage (Actuator): the stage component. The stage has to have metadata "MD_POS_COR" and "MD_FAV_FM_POS_ACTIVE"
-    return (dict str->float): the transformed position. It returns the updated axes x, y, rx, rz. The axis z is same as the input.
+    return (dict str->float): the transformed position. It returns the updated axes x, y, m, z, rx, rm. The axis z is same as the input.
     """
-    if not {"x", "y", "rz", "rx"}.issubset(stage.axes):
-        raise KeyError("The stage misses 'x', 'y', 'rx' or 'rz' axes")
+    if not {"x", "y", "m", "z", "rx", "rm"}.issubset(stage.axes):
+        raise KeyError("The stage misses 'x', 'y', 'm', 'z', 'rx' or 'rm' axes")
```

Comment:
```suggestion
    if not {"x", "y", "m", "z", "rx", "rm"}.issubset(stage.axes):
        raise KeyError("The stage misses %s axes" %({'x', 'y', 'm', 'z', 'rx', 'rm'} - stage.axes.keys()))
```

Reply:
It does not works well with the calculation in the message, so I have done a bit differently
`    if not {"x", "y", "m", "z", "rx", "rm"}.issubset(stage.axes):
        missed_axes = {'x', 'y', 'm', 'z', 'rx', 'rm'} - stage.axes.keys()
        raise KeyError("The stage misses %s axes" % missed_axes)`